### PR TITLE
Refactor js modules

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

Been working on decoupling the shared component library, which lead me down a rabbit hole of trying to decouple fonts, which lead to some refactoring of user-agent stuff and other utility modules.

This PR should have no functional changes

### Reviewer guidance

* renamed plugin from `user_agent` to `frontend_head_assets`
* renamed `plugin_data` to `plugin_data_src` to prevent mistakenly importing it
* tried re-organizing browser info, redirecting, font loading, and other logic

feedback is welcome

### References

N/A

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/kolibri_plugin.py [hunk 1]
```diff
@@ -96,7 +96,17 @@ def render_to_page_load_sync_html(self):
...
+        static_root = static("assets/fonts/noto-full")
+        full_file = "{}.{}.{}.css?v={}"
         return {
+            "fullCSSFileModern": full_file.format(
+                static_root, language_code, "modern", kolibri.__version__
+            ),
+            "fullCSSFileBasic": full_file.format(
+                static_root, language_code, "basic", kolibri.__version__
+            ),
+            "unsupportedUrl": reverse("kolibri:core:unsupported"),
```
Comment (rtibbles) [inline kolibri/core/kolibri_plugin.py]: This needs to be associated with the bundle that it was originally associated with - this variable is used by the browser minimum requirements check.

**Review by rtibbles — DISMISSED** [review]
One thing that I think has to be changed, other changes seem reasonable.

The main thing I would want to check is that tree-shaking is happening as intended, otherwise, this could bloat the `user_agent`/`frontend_head_requirements` with `Screenfull` as an unnecessary dependency.

**Comment by indirectlylit** [comment]
Ready for re-review.

> This needs to be associated with the bundle that it was originally associated with - this variable is used by the browser minimum requirements check.

Good catch - fixed.

> The main thing I would want to check is that tree-shaking is happening as intended, otherwise, this could bloat the user_agent/frontend_head_requirements with Screenfull as an unnecessary dependency.

You're right, `Screenfull` was being added to both `default_frontend` and `frontend_head_assets`. I've updated the PR to fix this.

Follow-up notes here: https://github.com/learningequality/kolibri/pull/6241

**Comment by indirectlylit** [comment]
will need to integrate changes from https://github.com/learningequality/kolibri/pull/6270

**Review by nucleogenesis — APPROVED** [review]
Very strange - it looks good to me in the code review but when I went to smoke test it I cannot build the frontend assets.

Everywhere that you replaced `kolibri.utils.browser` with `kolibri.utils.browserInfo` except `CoreFullscreen.vue` isn't being pulled into my local repo for some reason. My HEAD is pointing to the right commit and I've git fetched everything a few times but those changes are not reflected in my local code.

While that is a strange issue indeed - the changes look good. When I grep-replace my code with the changes that are certainly in this commit, everything passes a smoke test.

**Comment by indirectlylit** [comment]
thanks - due to botched merge

**Pull request merged**